### DISCUSSION TITLE
How to Reset the Page Number on Search

### DISCUSSION THREAD

**Discussion by kampit** [livewire/livewire#8597] [1. Help] [answered]
```
<?php

use App\Models\Company;
use function Livewire\Volt\{computed, state, usesPagination};

usesPagination();

state(['search'])->url(history: true, keep: true);

$companies = computed(fn () => Company::when($this->search, fn ($query) => $query->where('name', 'like', '%' . $this->search . '%')
    ->orWhere('address1', 'like', '%' . $this->search . '%')
    ->orWhere('country', 'like', '%' . $this->search . '%'))
    ->paginate(10)
);

?>

<div>

    <input wire:model.live.debounce.300ms="search" type="search">

    @foreach ($this->companies as $company)
    <div> {{ $company->name }}</div>
    <div> {{ $company->address1 }}</div>
    <div> {{ $company->country }}</div>
    @endforeach

    {{ $this->companies->links() }}

</div>
```

I am experiencing an issue where the page number is not reset when performing a new search. For example, if I am on page 10 of the dashboard:

`http://127.0.0.1:8000/dashboard?page=10`

And then I search for a keyword (e.g., hello), the URL becomes:

`http://127.0.0.1:8000/dashboard?page=10&search=hello`

I want the page number to reset to 1 (or remove the page query parameter) when a new search is performed, so the URL should be:

`http://127.0.0.1:8000/dashboard?search=hello`

How can I achieve this behavior in Livewire Volt when using `usesPagination`?

**Comment by rusdyahmad** [comment, marked as answer]
Refer [`$this->resetPage()`](https://livewire.laravel.com/docs/pagination#resetting-the-page)

When a new search is performed, the query builder first applies the search constraints using `$this->search`, and then the `tap` method is used to reset the page number to 1 before paginating the results.

So something like this

```
$companies = computed(fn () => Company::query()
    ->when($this->search, fn ($query) => $query->where('name', 'like', '%' . $this->search . '%')
        ->orWhere('address1', 'like', '%' . $this->search . '%')
        ->orWhere('country', 'like', '%' . $this->search . '%')
        ->tap(fn () => $this->resetPage()))
    ->paginate(10)
);
```

**Reply by kampit** [reply]
I couldn't move to the next page, but fixing it was a big help. I just moved `$this->resetPage()` outside, and now everything works smoothly. Thanks!

`$updatedSearch = fn() => $this->resetPage();`
